classes/02class/exercises/c02-network07/zarajoy/README.md

```diff
@@ -0,0 +1,445 @@
+z@bacon:~$ aws ec2 allocate-address --domain vpc
+{
+    "PublicIp": "13.236.53.110",
+    "AllocationId": "eipalloc-0a43fcd6fdd33cd2a",
+    "PublicIpv4Pool": "amazon",
+    "NetworkBorderGroup": "ap-southeast-2",
+    "Domain": "vpc"
+}
+z@bacon:~$ aws ec2 create-nat-gateway --subnet-id subnet-00649d5b0632cc2be --allocation-id eipalloc-0a43fcd6fdd33cd2a
+{
+    "NatGateway": {
+        "CreateTime": "2020-07-02T01:22:32.000Z",
+        "NatGatewayAddresses": [
+            {
+                "AllocationId": "eipalloc-0a43fcd6fdd33cd2a"
+            }
+        ],
+        "NatGatewayId": "nat-0f9b9a9f49bb2ddd7",
+        "State": "pending",
+        "SubnetId": "subnet-00649d5b0632cc2be",
+        "VpcId": "vpc-09d2f2719d50d1f7f"
+    }
+}
+z@bacon:~$ aws ec2 create-tags --resources nat-0f9b9a9f49bb2ddd7 --tags Key=Name,Value=DevopsAcademy-ngw
+z@bacon:~$ aws ec2 describe-nat-gateways 
+{
+    "NatGateways": [
+        {
+            "CreateTime": "2020-07-02T01:22:32.000Z",
+            "NatGatewayAddresses": [
+                {
+                    "AllocationId": "eipalloc-0a43fcd6fdd33cd2a",
+                    "NetworkInterfaceId": "eni-02cd3777735d4f4df",
+                    "PrivateIp": "192.168.168.93"
+                }
+            ],
+            "NatGatewayId": "nat-0f9b9a9f49bb2ddd7",
+            "State": "pending",
+            "SubnetId": "subnet-00649d5b0632cc2be",
+            "VpcId": "vpc-09d2f2719d50d1f7f",
+            "Tags": [
+                {
+                    "Key": "Name",
+                    "Value": "DevopsAcademy-ngw"
+                }
+            ]
+        }
+    ]
+}
+z@bacon:~$ aws ec2 create-route  --route-table-id rtb-0f8fb399d2a29b2fa  --destination-cidr-block 0.0.0.0/0 --nat-gateway-id nat-0f9b9a9f49bb2ddd7
+{
+    "Return": true
+}
+
+```
+- Answer the following questions:
+
+  -How did you accomplish both instructions?
+    ```
+   attached existing IGW - recreated NATGW as shown in commands
+    ```
+
+  Why did you add 0.0.0.0 route to the IGW on public route table?
+    ```
+    The destination for the route is 0.0.0.0/0, which represents all IPv4 addresses
+    ```
+
+  Why did you add 0.0.0.0 route to the NGW on private route table?
+    ```
+    like the reasoning for the 0.0.0.0/0 route on iGW, adding 0.0.0.0/0 on the NATGW allows private resources access to all of the ipv4 addesses
+    ```
+    
+  What is the difference of IGW and NGW?
+    ```
+    An Internet Gateway allows resources within your VPC to access the internet, and the internet to access your public subnet (and its resources)
+
+  NAT Gateway allows resources in a private subnet to access the internet.
+  It only works one way. The internet at large cannot get through your NAT to your private resources unless you explicitly allow it. A NAT gateway only exists in a single AZ so for all our 3 private subnets to access internet, we must have a NAT in each AZ.
```

Comment:
with a single NAT gateway all 3 private subnets can reach the Internet. 

The reason for having more than one NAT gateway is for having HA in case an AZ goes down.

Can you fix this one?